src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -443,20 +455,54 @@ func (l *commitLog) write() {
 			}
 		}
 
-		err := l.writerState.writer.Write(write.series,
-			write.datapoint, write.unit, write.annotation)
+		var (
+			numWritesSuccess int64
+			numDequeued      int
+		)
 
-		if err != nil {
-			l.metrics.errors.Inc(1)
-			l.log.Errorf("failed to write to commit log: %v", err)
+		if write.write.writeBatch == nil {
+			// Handle individual write case
+			numDequeued = 1
+			write := write.write.write
+			err := l.writerState.writer.Write(write.Series,
+				write.Datapoint, write.Unit, write.Annotation)
+			if err != nil {
+				l.handleWriteErr(err)
+				continue
```

Comment:
Hm do we really want to continue here?  We won't dequeue the num writes in queue if so.

Should we add a test to catch this too perhaps?  Or find some way to have a single code path?

Reply:
Sync'd offline, found a single code path solution